ql/src/java/org/apache/hadoop/hive/ql/udf/generic/GenericUDF.java

```diff
@@ -351,8 +352,12 @@ protected void obtainIntConverter(ObjectInspector[] arguments, int i,
     case INT:
     case VOID:
       break;
+    case DECIMAL:
+      if (inOi.scale() == 0) {
+        break;
+      }
```

Comment:
I think `precision` should also be small enough to ensure type safety. If we accept any DECIMAL, we should semantically accept LONG just in my opinion.

Reply:
Agreed

Reply:
Yes, I agree that we should limit precision. Since we are using `add(int field, int amount)` method of `java.util.Calendar`, where the `amount` type is an `int`, I think we should limit the value to max value of an int.